src/docs/about.md

```diff
@@ -0,0 +1,186 @@
+
+This should all be fairly familiar to you if you've ever written JavaScript (or any programming language for that
+matter) before, it's just inputs and output.
+
+## Thinking in Broccoli
+
+There are 2 main concepts to get your head around when using Broccoli:
+
+* [Plugins](#plugins)
+* [Trees](#trees)
+
+### Plugins
+
+Plugins are what a build pipeline developer will interact with most. Plugins are what do the actual work of
+transforming input files into output files. The API of a plugin is pretty simple, all that's required is creating
+a class that extends the [broccoli-plugin](https://github.com/broccolijs/broccoli-plugin) base class, and
+implementing a `build()` method, that performs some work or returns a promise.
+
+```js
+const Plugin = require('broccoli-plugin');
+
+class MyPlugin extends Plugin
+{
+  build() {
+    // A plugin can receive single or multiple inputs nodes/directories
+    // The files are available in this.inputPaths[0]/this.inputPaths[1]...
+    // You can do anything with these files that you can do in node
+    // The output of your plugin must write to this.outputPath
+  }
+}
+```
+
+A broccoli-plugin has only 1 purpose to transform the files from `this.inputPaths` directories to their output
+directory in this.outputPath directory when its `build()` function is invoked. Anything you can do in node, 
+you can do in the `build()` method. A plugin can receive one or multiple inputs, and these are available in the 
+`this.inputPaths` array in the order they are provided. `this.inputPaths` contains paths to directories, that are
+the `outputPath` of previous plugins. Each `inputPath` contains files that you can manipulate and write to
+`this.outputPath`. Broccoli will handle the state of these directories and passing them between plugins.
+
+There is a special case where a `string` is passed as an input to a plugin. When parsing your build pipeline, 
+Broccoli will automatically convert a string input into a
+[source plugin](https://github.com/broccolijs/broccoli-source). This plugin basically connects its input directory
+directly to its output directory, and also allows Broccoli to `watch` and be notified when files within the input
+directory change and trigger a rebuild. You can also manually create an `unwatched` directory from a string by
+using [UnwatchedDir](https://github.com/broccolijs/broccoli-source#new-unwatcheddirdirectorypath-options).
+
+### Trees
+
+Working from the target directory (the final output) back up to the source directories resembles a tree. Think 
+of a piece of Broccoli and you should have a mental model of what a build pipeline looks like.
+
+Broccoli constructs this tree of connected nodes in memory as it parses the `Brocfile`, and sets up the filesystem 
+state for each node, creating an `outputPath` for each node to write to.
+
+You may often encounter the term "tree" when reading plugin READMEs or in tutorials, just remember a tree is a
+connected set of plugins/nodes.
```

Comment:
One of the important concepts to get across is that trees can be 

- a string
- something a plugin emits